lib/cisco_node_utils/yum.rb

```diff
@@ -66,27 +66,78 @@ def self.detect_vrf
     end
 
     def self.install(pkg, vrf=nil)
-      vrf = vrf.nil? ? detect_vrf : "vrf #{vrf}"
-      config_set('yum', 'install', pkg, vrf)
+      if "#{platform}" == 'nexus'
+        vrf = vrf.nil? ? detect_vrf : "vrf #{vrf}"
+        config_set('yum', 'install', pkg, vrf)
 
-      # HACK: The current nxos host installer is a multi-part command
-      # which may fail at a later stage yet return a false positive;
-      # therefore a post-validation check is needed here to verify the
-      # actual outcome.
-      validate(pkg)
+        # HACK: The current nxos host installer is a multi-part command
+        # which may fail at a later stage yet return a false positive;
+        # therefore a post-validation check is needed here to verify the
+        # actual outcome.
+        validate(pkg)
+
+      elsif "#{platform}" == 'ios_xr'
+        filename = pkg.strip.tr(':', '/').split('/').last
+        name_arch_regex_xr = /^(.*\d.*-[\d.]*-r\d\d\d\d*.).(\w{4,}).rpm/
+        if filename =~ name_arch_regex_xr
+          pkg_name = Regexp.last_match(1)
+          debug "Installing package #{pkg_name}"
+          `ip netns exec default sdr_instcmd install activate \
+    pkg 0x0 #{pkg_name}`
+          `ip netns exec default sdr_instcmd install commit sdr`
+          # debug "#{rc}"
+        else
+          fail 'Failed to install the requested rpm.'
+        end
+      end
     end
 
     # returns version of package, or false if package doesn't exist
-    def self.query(pkg)
-      fail TypeError unless pkg.is_a? String
-      fail ArgumentError if pkg.empty?
-      b = config_get('yum', 'query', pkg)
-      fail "Multiple matching packages found for #{pkg}" if b && b.size > 1
+    def self.query(pkg, src)
+      if "#{platform}" == 'nexus'
+        fail TypeError unless pkg.is_a? String
+        fail ArgumentError if pkg.empty?
+        b = config_get('yum', 'query', pkg)
+        fail "Multiple matching packages found for #{pkg}" if b && b.size > 1
+
+      elsif "#{platform}" == 'ios_xr'
+        filename = src.strip.tr(':', '/').split('/').last
+        name_arch_regex_xr = /^(.*\d.*-[\d.]*-r\d\d\d\d*.).(\w{4,}).rpm/
```

Comment:
This is a pretty gnarly regexp, and you're declaring it in two different places. Can you make it a constant and add comments showing examples of what it should and should not be matching?